sdk-setup/src/mb2

```diff
@@ -1389,6 +1450,81 @@ run_prepare() {
     return $rc
 }
 
+cmake_post_cleanup() {
+    # Handle path changes for Qt Creator
+    local cmakefiles="CMakeCache.txt"
+
+    cp -p CMakeCache.txt CMakeCache.txt.qtccache
+
+    for cbpfile in *.cbp; do
+        [ -f "$cbpfile" ] || continue
+        cp -p "$cbpfile" "$cbpfile.qtccache"
+        cmakefiles="$cmakefiles $cbpfile"
+    done
+
+    if [[ "$STATEDIR" == "/home/mersdk/share"* && $MER_SSH_HOST_SHARED_HOME ]]; then
+        sed -i "s|/home/mersdk/share|$MER_SSH_HOST_SHARED_HOME|g" $cmakefiles
+    elif [[ "$STATEDIR" == "/home/src1"* && $MER_SSH_HOST_SHARED_SRC ]]; then
+        sed -i "s|/home/src1|$MER_SSH_HOST_SHARED_SRC|g" $cmakefiles
+    else
+        echo "Failed to set CMake paths for Qt Creator"
+        rm *.qtccache
+    fi
+}
+
+cmake_pre_setup() {
+    # Undo path changes for Qt Creator
+    if [[ -f "CMakeCache.txt.qtccache" ]]; then
+        mv CMakeCache.txt.qtccache CMakeCache.txt
+        for cbp_file in *.cbp.qtccache; do
+            mv $cbp_file ${cbp_file%.qtccache}
+        done
+    fi
+}
+
+run_cmake() {
+    local rpmbuild="--build . --target all"
+    local sb2args=( -t "$OPT_TARGET" env PATH="${WRAPPERSDIR}:${USERWRAPPERSDIR}:${PATH}" )
+
+    # CMake projects need to have dependencies installed before running
+    # any CMake commands
+    if [[ $OPT_SPEC ]]; then
+        verify_target_dependencies
+    fi
+
+    if [[ " $* " == *" $rpmbuild "* && $OPT_SPEC ]]; then
+        local build_tgt=$(sb2 -t $OPT_TARGET gcc -dumpmachine)
+        build_tgt=${build_tgt%-gnueabi}
+
+        cp -a "$OPT_SPEC" "$TMP_SPEC"
+
+        # get rid of potential windows linefeeds
+        sed -i "s/\x0D$//g" "$TMP_SPEC"
+
+        fix_package_version "$TMP_SPEC"
+
+        eval "set -- $(printf "%q " "$@" |sed "s/ $rpmbuild / /")"
+        init_cmake_wrapper $WRAPPERSDIR/cmake "$@" "$OPT_SRC_DIR"
+        init_simple_wrapper $WRAPPERSDIR/make
+
+        (
+            cd_to_spec_setup_dir
+            
+            echo sb2 "${sb2args[@]}" rpmbuild --build-in-place --target=$build_tgt --define "noecho 1" \
+                --define "_sourcedir $(dirname $OPT_SPEC)" -bc --short-circuit "$TMP_SPEC"
+            
+            sb2 "${sb2args[@]}" rpmbuild --build-in-place --target=$build_tgt --define "noecho 1" \
+                --define "_sourcedir $(dirname $OPT_SPEC)" -bc --short-circuit "$TMP_SPEC"
+    
+        )
+    else
+        init_cmake_wrapper $WRAPPERSDIR/cmake "$@"
+        sb2 "${sb2args[@]}" cmake
```

Comment:
Is the wrapper really needed here? Maybe the following is enough as it is in the qmake case:

`sb2 -t $OPT_TARGET cmake "$@" ${OPT_PRO_FILE_OR_DIR:+"$OPT_PRO_FILE_OR_DIR"}`

In any case the `-t` option is missing.

Reply:
I think you are right.